python/filtering/filter_item_widget.py

```diff
@@ -91,19 +99,31 @@ def value(self):
     def value(self, value):
         raise sgtk.TankError("Abstract class method not overriden")
 
+    @property
+    def sort_value(self):
+        """Get the widget's value that is sortable."""
+        # First try to extract the sort value from the raw value.
+        value = self._raw_value
+        if isinstance(value, dict):
+            value = value.get("name")
+        # Default to the display name if the raw value is not sortable.
+        if value is None:
```

Comment:
You might prefer `if not value` which will match with both None value and empty string.

Reply:
The empty string is ok here, we want to use the raw value to sort as long as it is a sortable value.